plink_clump_hail.py

```diff
@@ -0,0 +1,366 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Mar  4 15:09:04 2020
+
+Hail script for clumping GWAS results with PLINK
+
+@author: nbaya
+"""
+
+import argparse
+import hail as hl
+import sys
+import ukb_common
+from ukbb_pan_ancestry import get_clumping_results_path #get_pheno_manifest_path
+#from ukb_common import mwzj_hts_by_tree
+
+bucket = 'gs://ukb-diverse-pops'
+ldprune_dir = f'{bucket}/ld_prune'
+
+all_pops = ['AFR', 'AMR', 'CSA', 'EAS', 'EUR', 'MID']
+
+def ht_to_tsv(args):
+    r'''
+    Convert Hail table of variant results to a tsv 
+    '''
+    add_args = {}
+    if args.n_threads is not None:
+        add_args['master'] = f'local[{args.n_threads}]'
+    hl.init(default_reference='GRCh38', log='/tmp/ht_to_tsv.log', **add_args)
+    ht = hl.read_table(args.input_file)
+    print(ht.describe())
+    ht.export(args.output_file)
+
+    
+def tsv_to_ht(args):
+    r'''
+    Convert tsv to a Hail table
+    '''
+    print(sys.version)
+    add_args = {}
+    if args.n_threads is not None:
+        add_args['master'] = f'local[{args.n_threads}]'
+    hl.init(default_reference='GRCh38', log='/tmp/tsv_to_ht.log', **add_args)
+    ht = hl.import_table(args.input_file, no_header=True, impute=False, 
+                         types={'f0':hl.tstr, 
+                                'f1':hl.tint, # casting to non-str type doesn't work if there are empty strings (as there may be in some of the existing clump_results.txt files)
+                                'f2':hl.tstr,
+                                'f3':hl.tint, 
+                                'f4':hl.tfloat,
+                                'f5':hl.tint,
+                                'f6':hl.tint,
+                                'f7':hl.tint,
+                                'f8':hl.tint,
+                                'f9':hl.tint,
+                                'f10':hl.tint,
+                                'f11':hl.tstr})
+    ht = ht.rename({'f0':'contig',
+                    'f1':'F',
+                    'f2':'varid',
+                    'f3':'pos',
+                    'f4':'P',
+                    'f5':'TOTAL',
+                    'f6':'NSIG',
+                    'f7':'S05',
+                    'f8':'S01',
+                    'f9':'S001',
+                    'f10':'S0001',
+                    'f11':'SP2'})
+
+    ht = ht.filter(ht.contig!='') # added because there were some files with extra lines with empty strings
+    ht = ht.order_by('P')
+    ht = ht.add_index()
+    ht = ht.key_by(locus = hl.locus(contig=ht.contig, 
+                                    pos=ht.pos,
+                                    reference_genome='GRCh37'),
+                   alleles = hl.array([ht.varid.split(':')[2],
+                                       ht.varid.split(':')[3]])
+    )
+    ht = ht.annotate_globals(trait_type = args.trait_type,
```

Comment:
`.annotate_globals(**{k: getattr(args, k) for k in PHENO_KEY_FIELDS})` if you import it